File: 13. Roman to Integer/romantoint.py
class Solution(object):
    def romanToInt(self, s):
        """
        :type s: str
        :rtype: int
        """
        symbol = {'I':1,'V':5,'X':10,'L':50,'C':100,'D':500,'M':1000}
        temp = [0]

        if "IV" in str(s):
          s = s.replace("IV","VIIII",1)
          print(s, "IV")
        if "IX" in str(s):
          s = s.replace("IX","VIIII",1)
          print(s, "IX")
        if "XL" in str(s):
          s = s.replace("XL","XXXX",1)
          print(s, "XL")
        if "XC" in str(s):
          s = s.replace("XC","LXXXX",1)
          print(s, "XC")
        if "CD" in s:
          s = s.replace("CD","CCCC",1)
          print(s, "CD")
        if "CM" in s:
          s = s.replace("CM","DCCCC",1)
          print(s, "CM")

        for i in range(len(s)):
            print(i)
            if s[i] == 'I':
                temp.insert(i, 1)
                print(s[i], temp)

            elif s[i] == 'V':
                temp.insert(i, 5)
                print(s[i], temp)

            elif s[i] == 'X':
                temp.insert(i,10)
                print(s[i], temp)

            elif s[i] == 'L':
                temp.insert(i,50)
                print(s[i], temp)

            elif s[i] == 'C':
                temp.insert(i,100)
                print(s[i], temp)

            elif s[i] == 'D':
                temp.insert(i,500)
                print(s[i], temp)

            elif s[i] == 'M':
                temp.insert(i,1000)
                print(s[i], temp)
            else:
                break
                
        return sum(temp)

File: 13. Roman to Integer/romantoint.js
/**
 * @param {string} s
 * @return {number}
 */
var romanToInt = function (s) {
  let symbol = { I: 1, V: 5, X: 10, L: 50, C: 100, D: 500, M: 1000 };
  let temp = [0];

  if (s.includes("IV")) {
    s = s.replace("IV", "IIII", 1);
    console.log(s, "IV");
  }
  if (s.includes("IX")) {
    s = s.replace("IX", "VIIII", 1);
    console.log(s, "IX");
  }
  if (s.includes("XL")) {
    s = s.replace("XL", "XXXX", 1);
    console.log(s, "XL");
  }
  if (s.includes("XC")) {
    s = s.replace("XC", "LXXXX", 1);
    console.log(s, "XC");
  }
  if (s.includes("CD")) {
    s = s.replace("CD", "CCCC", 1);
    console.log(s, "CD");
  }
  if (s.includes("CM")) {
    s = s.replace("CM", "DCCCC", 1);
    console.log(s, "CM");
  }

  for (let i = 0; i < s.length; i++) {
    console.log(i);
    if (s[i] === "I") {
      temp.splice(i, 0, 1);
      console.log(s[i], temp);
    } else if (s[i] === "V") {
      temp.splice(i, 0, 5);
      console.log(s[i], temp);
    } else if (s[i] === "X") {
      temp.splice(i, 0, 10);
      console.log(s[i], temp);
    } else if (s[i] === "L") {
      temp.splice(i, 0, 50);
      console.log(s[i], temp);
    } else if (s[i] === "C") {
      temp.splice(i, 0, 100);
      console.log(s[i], temp);
    } else if (s[i] === "D") {
      temp.splice(i, 0, 500);
      console.log(s[i], temp);
    } else if (s[i] === "M") {
      temp.splice(i, 0, 1000);
      console.log(s[i], temp);
    } else {
      break;
    }
  }

  let result = temp.reduce((a, b) => a + b, 0);
  return result;
};
